File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    var answer=''
    let map = new Map()
    
    for(let i =0; i<participant.length; i++){
        if(!map.get(participant[i])){
            map.set(participant[i], 1)
        }
        else{
            map.set(participant[i], map.get(participant[i])+1)
        }
    }
    for(let temp of completion){
        if(map.get(temp)){
            map.set(temp, map.get(temp)-1);
        }
    }
    
     for(let i =0; i< participant.length; i++){
        if(map.get(participant[i])===1){
            answer = participant[i]
            break
        }
     }
         
    return answer
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;

class Solution {
    public String solution(String[] participant, String[] completion) {
        String answer = "";
        Map<String,Integer> map = new HashMap<>();
        
   
        //participant로 1씩 value값 키우기
        for(String data : participant){
            map.put(data, map.getOrDefault(data, 0) + 1);
        }
        
        //완주자 1씩 감소시키기
        for(String data : completion){
            map.put(data, map.getOrDefault(data, 0) - 1);
        }
        
    
        
        for (Map.Entry<String, Integer> entry : map.entrySet()) {
    if (entry.getValue() == 1) {
        answer+= entry.getKey();
    }
}
        return answer;
    }
}